awx/api/serializers.py

```diff
@@ -4230,6 +4235,67 @@ def get_summary_fields(self, obj):
         d['recent_notifications'] = self._recent_notifications(obj)
         return d
 
+    def validate_messages(self, messages):
+        if messages is None:
+            return None
+
+        error_list = []
+        collected_messages = []
+
+        if not isinstance(messages, dict):
+            error_list.append(_("Expected dict for 'messages' field, found {}".format(type(messages))))
+        else:
+            for event in messages:
+                if event not in ['started', 'success', 'error']:
+                    error_list.append(_("Event '{}' invalid, must be one of 'started', 'success', or 'error'").format(event))
+                    continue
+                event_messages = messages[event]
+                if event_messages is None:
+                    continue
+                if not isinstance(event_messages, dict):
+                    error_list.append(_("Expected dict for event '{}', found {}").format(event, type(event_messages)))
+                    continue
+                for message_type in event_messages:
+                    if message_type not in ['message', 'body']:
+                        error_list.append(_("Message type '{}' invalid, must be either 'message' or 'body'").format(message_type))
+                        continue
+                    message = event_messages[message_type]
+                    if message is None:
+                        continue
+                    if not isinstance(message, str):
+                        error_list.append(_("Expected string for '{}', found {}, ").format(message_type, type(message)))
+                        continue
+                    collected_messages.append(message)
+
+        # Subclass to return name of undefined field
+        class DescriptiveUndefined(StrictUndefined):
+            # The parent class prevents _accessing attributes_ of an object
+            # but will render undefined objects with 'Undefined'. This
+            # prevents their use entirely.
+            __repr__ = __str__ = StrictUndefined._fail_with_undefined_error
+
+            def __init__(self, *args, **kwargs):
+                super(DescriptiveUndefined, self).__init__(*args, **kwargs)
+                # When an undefined field is encountered, return the name
+                # of the undefined field in the exception message
+                # (StrictUndefined refers to the explicitly set exception
+                # message as the 'hint')
+                self._undefined_hint = self._undefined_name
+
+        # Ensure messages can be rendered
+        for msg in collected_messages:
+            env = sandbox.ImmutableSandboxedEnvironment(undefined=DescriptiveUndefined)
```

Comment:
@keithjgrant I think we should adjust the UI to surface this error somehow in the UI:

![Kapture 2019-08-15 at 17 04 23](https://user-images.githubusercontent.com/214912/63126830-cbf31200-bf7e-11e9-9e57-d5db66bdd393.gif)

Reply:
my opinion is that this is one necessary change as a requirement to merge cc @wenottingham for his opinion

Reply:
oh, good call, @ryanpetrello - I noticed this the other day, but failed to follow up. @keithjgrant - the UI used to surface the actual error message (remember seeing this when I first demo-ed the branch). Do you know what changed?

Reply:
I concur. I've added a patch for this to the feature branch.
![Screenshot from 2019-08-16 10-16-11](https://user-images.githubusercontent.com/9753817/63174874-d827af80-c010-11e9-922f-d58ddb730093.png)

@unlikelyzero is this scenario something you can incorporate into your automated tests?